tests/serialization_test.py

```diff
@@ -173,6 +174,9 @@ def test_numpy_serialization(self):
                      'complex', 'bool']
     np.random.seed(0)
     for dtype in normal_dtypes:
+      if (dtype in {'float128', 'f16', 'complex256', 'c32'}) and (platform.system() == 'Darwin') and (platform.machine() == 'arm64'):
```

Comment:
Perhaps we should print a warning so the user is aware that these tests are being skipped?

Reply:
Maybe we should use the `pytest.mark.skipif` decorator over the test?